go/enclave/core/batchlimiter.go

```diff
@@ -0,0 +1,80 @@
+package core
+
+import (
+	"errors"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+type BatchSizeLimiter struct {
+	Size            uint64
+	ContractAddress common.Address
+	Topic           common.Hash
+}
+
+var ErrInsufficientSpace = errors.New("insufficient space in BatchSizeLimiter")
+
+func NewBatchSizeLimiter(size uint64, contractAddress common.Address, topic common.Hash) *BatchSizeLimiter {
+	return &BatchSizeLimiter{
+		Size:            size,
+		ContractAddress: contractAddress,
```

Comment:
Was a bit confused why a batch size limiting thing would need a contract address and a topic when it could have all sorts of transactions in it. Seems they're related to the bridge stuff though? So could maybe rename the param or add a comment to clarify.